kale/evaluate/cross_validation.py

```diff
@@ -0,0 +1,57 @@
+import numpy as np
+from sklearn.metrics import accuracy_score
+from sklearn.preprocessing import OneHotEncoder
+
+
+def leave_one_out_cross_validate(x, y, covariates, estimator, domain_adaptation=False) -> dict:
+    """
+    Perform leave-one-out cross validation (LOOCV) for a given estimator.
+
+    Args:
+        x: Input data [n_samples, n_features].
+        y: Target labels [n_samples].
+        covariates: Covariates indicating classes for LOOCV [n_samples].
+        estimator: Machine learning estimator to be evaluated from kale or scikit-learn.
+        domain_adaptation: Whether to use domain adaptation during training.
+
+    Returns:
+        dict: A dictionary containing LOOCV results for each target group with 3 keys.
+
+            - 'Target': List of unique target groups or classes.
+            - 'Num_samples': List containing number of samples in each target group.
+            - 'Accuracy': List of accuracy scores for each target group.
+    """
+    target, num_samples, accuracy = [], [], []
+    enc = OneHotEncoder(handle_unknown="ignore")
+    covariate_mat = enc.fit_transform(covariates.reshape(-1, 1)).toarray()
+    unique_covariates = np.unique(covariates)
+
+    for tgt in unique_covariates:
```

Comment:
Try if the [LeaveOneGroupOut API](https://scikit-learn.org/stable/modules/cross_validation.html#leave-one-group-out) in scikit-learn can be leveraged here.